frontend/src/features/navigation/header/accountMenu.tsx

```diff
@@ -0,0 +1,100 @@
+import { KeyboardArrowDownOutlined, KeyboardArrowUpOutlined } from '@mui/icons-material';
+
+import Button from '@mui/material/Button';
+import ClickAwayListener from '@mui/material/ClickAwayListener';
+import Grow from '@mui/material/Grow';
+import { Link } from 'react-router-dom';
+import MenuItem from '@mui/material/MenuItem';
+import MenuList from '@mui/material/MenuList';
+import Paper from '@mui/material/Paper';
+import Popper from '@mui/material/Popper';
+import React from 'react';
+import { styled } from '@mui/material/styles';
+import { useAuth } from '../../../providers/auth-provider';
+
+const menuItems = [
+  { label: 'Saved recipes', path: '/recipes/saved' },
+  { label: 'My recipes', path: '/recipes' },
+  { label: 'Log out', path: '' },
+];
+
+const AccountMenu = () => {
+  const [open, setOpen] = React.useState(false);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const { logout } = useAuth();
+
+  const handleOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleMenuItemClick = (label: string) => {
+    if (label === 'Log out') {
```

Comment:
Can we have better typing here? What if the label changes on the button? the functionality would stop working